scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/context/TaskContextVariableExtractor.java

```diff
@@ -120,6 +120,27 @@
         return variables;
     }
 
+    private Map<String, Serializable> addTaskVariablesAndBackupInheritedVariables(Map<String, Serializable> variables,
+            ImmutableMap<String, TaskVariable> taskVariables) {
+        Map<String, Serializable> taskAndBackupVariables = new HashMap<>();
+        for (TaskVariable taskVariable : taskVariables.values()) {
+            if (!taskVariable.isJobInherited() ||
+                (taskVariable.isJobInherited() && variables.get(taskVariable.getName()) == null)) {
```

Comment:
You should create a local variable to hold the result of `taskVariable.getName()` since it is used many times.

Reply:
definitely, thanks for the remark!